enhanced documentation on adyen deployment on cloud infrastructure

JudeNiroshan opened the pull request:
## Description

- Moved k8s yaml files in to deployment-examples directory
- Remove redundant stuff module specific documentation files.

ahmetoz commented on README.md:
```diff
@@ -1,3 +1,16 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**  
```
I would put this after the badges and main description of this repository (before overview header), also the readme is not so long, so not much value in TOC (imo)

JudeNiroshan replied:
591abed05b49799ae0dcb6300ea7d497eb0c1323

ahmetoz commented on deployment-examples/aws/extension/README.md:
```diff
@@ -0,0 +1,13 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**  
```
I found TOCs valuable on when the things are complex and needs navigation, in this document it's quite easy to see what is this about, is not it ?

JudeNiroshan replied:
41f888c3f1faf527e3e798aec08984fa7077d7e7

ahmetoz commented on deployment-examples/aws/notification/README.md:
```diff
@@ -0,0 +1,12 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**  
```
same here.

JudeNiroshan replied:
41f888c3f1faf527e3e798aec08984fa7077d7e7

ahmetoz commented on deployment-examples/gcp/README.md:
```diff
@@ -1,21 +1,29 @@
+**Table of Contents**  
+<!-- *generated with [DocToc](https://github.com/thlorenz/doctoc)* -->
+
+- [Continuous Deployment Guide](#continuous-deployment-guide)
```
same here.

JudeNiroshan replied:
41f888c3f1faf527e3e798aec08984fa7077d7e7

ahmetoz commented on docs/ContributionGuide.md:
```diff
@@ -1,16 +1,17 @@
 <!-- START doctoc generated TOC please keep comment here to allow auto update -->
 <!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
-**Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*
+**Table of Contents**  
```
I could not get the benefit of changing the existing development guides 🤔 Looks harder to maintain than before (now you need to specify like this is only for extension module)

JudeNiroshan replied:
yeah, you are correct. But at this moment, if someone wants to make PR, all the information to setup the project locally is there. I think at some point we need to separate the files, but by looking at them right now, one contribution guide for the repository makes sense for me.

ahmetoz commented on deployment-examples/gcp/README.md:
```diff
@@ -1,21 +1,29 @@
 
 As a quick start, you can utilize the available bash script which is capable of deploying both modules to a 
-Google Cloud Project. It requires to have:
+public cloud provider.
```
it's not generic, so specify clearly that this guide only makes sense for GCP.

JudeNiroshan replied:
e41146b384d6cb54e2b6dda625b946ffc8c92558

ahmetoz commented on deployment-examples/gcp/README.md:
```diff
@@ -1,21 +1,29 @@
-3. Ensure the `secrets.yaml` files ([extension](./../extension/k8s/demo/secrets.yaml) and [notification](./../notification/k8s/demo/secrets.yaml)) have encrypted using the `GCLOUD_KMS_KEY_NAME` key.
+- An existing Kubernetes cluster in [Google Kubernetes Engine](https://cloud.google.com/kubernetes-engine/docs/quickstart)
+- Configure both module specific configurations in yaml files that resides under `k8s` folders.
+    Refer to [`extension/docs/DeploymentGuide.md`](../../extension/docs/HowToRun.md) and [`notification/docs/DeploymentGuide.md`](../../notification/docs/HowToRun.md) docs.
```
Honestly, this is hard to get the problem in old documents, what was wrong in the old DeploymentGuide file :D even the contents are the same, why not just referencing the deployment examples in our docs?

I think it's not good to change file names as it's a breaking change for people who has the links :)

JudeNiroshan replied:
>what was wrong in the old DeploymentGuide file 

We should not confuse the reader on executing the module in their machines vs. deploying the infrastructure for production usage. Therefore, I thought not to use "deployment" word in multiple places, rather gather all information required for deployment in one place.

ahmetoz commented on deployment-examples/gcp/README.md:
```diff
@@ -1,21 +1,29 @@
-3. Ensure the `secrets.yaml` files ([extension](./../extension/k8s/demo/secrets.yaml) and [notification](./../notification/k8s/demo/secrets.yaml)) have encrypted using the `GCLOUD_KMS_KEY_NAME` key.
+- An existing Kubernetes cluster in [Google Kubernetes Engine](https://cloud.google.com/kubernetes-engine/docs/quickstart)
+- Configure both module specific configurations in yaml files that resides under `k8s` folders.
+    Refer to [`extension/docs/DeploymentGuide.md`](../../extension/docs/HowToRun.md) and [`notification/docs/DeploymentGuide.md`](../../notification/docs/HowToRun.md) docs.
```
```suggestion
    Refer to [`extension/docs/HowToRun.md`](../../extension/docs/HowToRun.md) and [`notification/docs/HowToRun.md`](../../notification/docs/HowToRun.md) docs.
```

JudeNiroshan replied:
3ddaaac940468fc2a91d40dbeca2216d3f8655b8

ahmetoz reviewed (changes requested):
My general overview

- the `deployment-examples` folder is a nice idea, cleaner than before but I would put the readme into that folder so when someone checks the examples folder directly could understand easily before navigating all folders.

- Could not reason the changes for the contribution guide, I think we should not combine development guides, imo it was better before, maybe you add it because of lots of duplicate content, I think it is not an issue, it's not a code though

- I think the deployment guide (how to run) does not explain much and does not navigate the exact places so additional navigation is needed. Also, I am missing the deployment examples related to content in the readme file, which could make sense to explain in the main readme.

- We need to manage version number in the deployment guide, I am seeing v5.0.0 as explained as a default tag which I am thinking will stay forever there :) maybe should not have a default value. Similar to the other files, you are saying default but the script file does not have those defaults, a bit confusing. (i.e : HELM_CHARTS_VERSION)

- IMO TOCs do not add value when the content is one page or even shorter.

JudeNiroshan commented:
>Could not reason the changes for the contribution guide, I think we should not combine development guides, imo it was better before, maybe you add it because of lots of duplicate content, I think it is not an issue, it's not a code though

The intuition behind this, people who use this project are mainly developers. When they see, development guide, they tend to think it is how to configure. And we also have another read me for integration guide. To clear out what exactly meant by "DevelopmentGuide", I moved it out and renamed it to "ContributionGuide" which is more clear for new developers who look at the repository for the first time.

ahmetoz commented on docs/ContributionGuide.md:
```diff
@@ -32,15 +33,16 @@ npm install
+#### E2E tests (Extension module only)
 Following additional environment variables must be provided in order to run the E2E tests.
+
 | Name | Content | Required | Default value |
```
Table markdown is not rendered correctly, can you fix it?

JudeNiroshan replied:
9eaad393597f7759b75eb39d244ce62dd76dbd0e

ahmetoz commented on extension/docs/HowToRun.md:
```diff
@@ -102,7 +100,8 @@ Refer to our [docker hub](https://hub.docker.com/r/commercetools/commercetools-a
 
-1. For deployment to AWS Lambda, zip the extensions folder and specify `src/lambda.handler` as the entry point for the AWS Lambda function.
-2. When the extension module is run as AWS Lambda **it will NOT create** the required resources like custom types or commercetools API extension for you. Please follow the [manual resource creation guide](#creating-resources-manually) instead. 
+Extension module supports different deployment [options](/deployment-examples). 
```
Could you link the GCP readme instead of linking to the deployment-examples directory.

JudeNiroshan replied:
but that will express the idea that we only support GCP deployment which is not true.

ahmetoz replied:
Then don't link the AWS lambda in here, the user can navigate, which does not make sense to me.

JudeNiroshan replied:
1ccf903e31fc2bb38b11f2dca34fc88d09da8879

ahmetoz commented on extension/docs/WebComponentsIntegrationGuide.md:
```diff
@@ -53,7 +53,7 @@ The following diagram shows checkout integration flow based on [Adyen Web Compon
 
 ## Before you begin
-In order to make the extension module up and running, follow our [deployment guide](./DeploymentGuide.md). For the sake of readability,
+In order to make the extension module up and running, follow our [deployment guide](docs/HowToRun.md). For the sake of readability,
```
```suggestion
In order to make the extension module up and running, follow our [how to run guide](docs/HowToRun.md). For the sake of readability,
```

JudeNiroshan replied:
794295beb59939654119977d146e51ab356019a4

ahmetoz commented on notification/docs/HowToRun.md:
```diff
@@ -64,6 +62,8 @@ Refer to our [docker hub](https://hub.docker.com/r/commercetools/commercetools-a
+## Deployment
 
-For deployment to AWS Lambda, zip the `notification` folder and specify `src/lambda.js` as the entry point for the AWS Lambda function.
+Notification module supports different deployment [options](/deployment-examples).
```
same here could you link to GCP readme ?

JudeNiroshan replied:
but that will express the idea that we only support GCP deployment which is not true.

JudeNiroshan replied:
1ccf903e31fc2bb38b11f2dca34fc88d09da8879

ahmetoz commented on notification/docs/IntegrationGuide.md:
```diff
@@ -29,7 +29,7 @@ Please follow the [instructions](https://docs.adyen.com/development-resources/we
 
 ## Step 2: Deploy the notification module
-In order to make the extension module up and running, follow our [deployment guide](./DeploymentGuide.md).
+In order to make the extension module up and running, follow our [deployment guide](docs/HowToRun.md).
```
```suggestion
In order to make the extension module up and running, follow our [how to run guide](docs/HowToRun.md).
```

JudeNiroshan replied:
794295beb59939654119977d146e51ab356019a4

ahmetoz commented on deployment-examples/gcp/README.md:
```diff
@@ -24,12 +22,12 @@ Google Cloud Project. It requires to have:
-|`TAG` | Indented [release version](https://github.com/commercetools/commercetools-adyen-integration/releases) of commercetools-adyen-integration | YES | `v5.0.0` |
+|`TAG` | Indented [release version](https://github.com/commercetools/commercetools-adyen-integration/releases) of commercetools-adyen-integration | YES | `x.x.x` |
 |`HELM_CHARTS_REPO` | Github repository URL for helm charts | NO | `https://github.com/commercetools/k8s-charts.git` |
 |`HELM_CHARTS_VERSION` | Intended release version of the helm charts repository | YES | `1.7.5` |
```
This is not a default in the script.

JudeNiroshan replied:
3c1ef8332db859f41467b4c5f390e70e35314f80

ahmetoz commented on deployment-examples/gcp/README.md:
```diff
@@ -24,12 +22,12 @@ Google Cloud Project. It requires to have:
 |`GCLOUD_KMS_KEYRING` | Google KMS [key-ring](https://cloud.google.com/kms/docs/resource-hierarchy#key_rings) which include the encrypted key for secret.yaml files | YES | `adyen-integration-deployment-demo` |
 |`GCLOUD_KMS_KEY_NAME` | Google KMS key name which used to encrypt the secret.yaml files | YES | `adyen-extension-module` |
-|`TAG` | Indented [release version](https://github.com/commercetools/commercetools-adyen-integration/releases) of commercetools-adyen-integration | YES | `v5.0.0` |
+|`TAG` | Indented [release version](https://github.com/commercetools/commercetools-adyen-integration/releases) of commercetools-adyen-integration | YES | `x.x.x` |
```
If default is not set, maybe the column should be empty, x.x.x does not make sense :)

JudeNiroshan replied:
7b99608d9f183390cd9fa58ed29b25ba761c22df

ahmetoz commented on deployment-examples/gcp/README.md:
```diff
@@ -24,12 +22,12 @@ Google Cloud Project. It requires to have:
 |`GCLOUD_CLUSTER_NAME` | Existing Google Kubernetes Engine cluster name | YES | `adyen-demo` |
```
cluster name should not have a default value. can you check both table and readme table and make them consistent to each other.

JudeNiroshan replied:
34666127b77bfce2ee7c41aae6209b6ae1020131

ahmetoz replied:
Having hard to time to diff when multiple comments addressed in one commit, I will review from stratch.

ahmetoz commented on deployment-examples/gcp/README.md:
```diff
@@ -24,12 +22,12 @@ Google Cloud Project. It requires to have:
 |`GCLOUD_KMS_KEY_NAME` | Google KMS key name which used to encrypt the secret.yaml files | YES | `adyen-extension-module` |
-|`TAG` | Indented [release version](https://github.com/commercetools/commercetools-adyen-integration/releases) of commercetools-adyen-integration | YES | `v5.0.0` |
+|`TAG` | Indented [release version](https://github.com/commercetools/commercetools-adyen-integration/releases) of commercetools-adyen-integration | YES | `x.x.x` |
 |`HELM_CHARTS_REPO` | Github repository URL for helm charts | NO | `https://github.com/commercetools/k8s-charts.git` |
```
HELM_CHARTS_REPO can not be changed, so maybe remove it from here? also, it's not an env variable imo :)

JudeNiroshan replied:
34666127b77bfce2ee7c41aae6209b6ae1020131

ahmetoz commented on deployment-examples/gcp/README.md:
```diff
@@ -24,12 +22,12 @@ Google Cloud Project. It requires to have:
 |`ENVIRONMENT_NAME` | Folder name that contains secrets.yaml file in `extension/k8s` and `notification/k8s` | YES | `demo` |
 
-After configured all the required values, execute the `deploy-to-gcp.sh` script file.
+After configured all the required values, execute the [`deploy-to-gcp.sh`](deploy-to-gcp.sh) script file.
```
```suggestion
After configured all the required environment variables, execute the [`deploy-to-gcp.sh`](deploy-to-gcp.sh) script file.
```

JudeNiroshan replied:
b3dcb6d4f7eda17c78d53582cc84983c9d35eec1

ahmetoz commented on deployment-examples/gcp/README.md:
````diff
@@ -24,12 +22,12 @@ Google Cloud Project. It requires to have:
 
-After configured all the required values, execute the `deploy-to-gcp.sh` script file.
+After configured all the required values, execute the [`deploy-to-gcp.sh`](deploy-to-gcp.sh) script file.
 ```
````
would you put the EXPORT statements for variables here to make it clear visually ( also could save time for the devs who would run the script) for the env variables? the table is something for descriptions imo :)

JudeNiroshan replied:
655af40a18f0b6b367f0a5011691acb3afbcb347

ahmetoz replied:
I'm not sure if this is what I meant.

JudeNiroshan replied:
fb884884da655d3f3dbd37070c836f9fa3449d5d

ahmetoz reviewed (approved):
Good work, I put some more comments, so it's up to you address.

JudeNiroshan commented on deployment-examples/gcp/README.md:
```diff
@@ -0,0 +1,35 @@
+- Configure the below values in [`deploy-to-gcp.sh`](deploy-to-gcp.sh) file.
+- Ensure the `secrets.yaml` files ([extension](extension/k8s/demo/secrets.yaml) and [notification](notification/k8s/demo/secrets.yaml)) have encrypted using the `GCLOUD_KMS_KEY_NAME` key.
+
+| Name | Description | Required |
```
9cf8aa6e0349e18546588f4f5c28754cf28240e4

JudeNiroshan commented on deployment-examples/gcp/deploy-to-gcp.sh:
```diff
@@ -2,24 +2,24 @@
 set -e
 
-echo "Login to gcloud SDK and select project"
+export GCLOUD_PROJECT_ID="xxx"
```
fb884884da655d3f3dbd37070c836f9fa3449d5d
